security/advancedtls/crl_provider.go

```diff
@@ -0,0 +1,229 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package advancedtls
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"time"
+)
+
+const defaultCRLRefreshDuration = 1 * time.Hour
+
+// CRLProvider is the interface to be implemented to enable custom CRL provider
+// behavior.
+//
+// The interface defines how the data is read, but doesn't prescribe a way
```

Comment:
Suggest something more along the lines of
"The interface defines how gRPC gets CRLs from the provider during handshakes, but doesn't prescribe a specific way to load and store CRLs."

Also suggest adding info about this being directly on the path of connection establishment, so implementations of the `CRL` function need to be fast, and slow things such as file IO should be done asynchronously.

Reply:
Done